lua/neogit/lib/git/diff.lua

```diff
@@ -31,17 +29,17 @@ local function parse_diff(output)
   for i=1,len do
     local line = hunks[i]
     if not vim.startswith(line, "+++") then
-      local matches = vim.fn.matchlist(line, "^@@ -\\([0-9]*\\),\\?\\([0-9]*\\)\\? +\\([0-9]*\\),\\?\\([0-9]*\\)\\? @@")
+      local index_from, index_len, disk_from, disk_len = line:match('@@ %-(%d+),?(%d*) %+(%d+),?(%d*) @@')
```

Comment:
Damn `string.match` looks way cleaner.

Reply:
You don't have to escape as much, but it's also less powerful than vim's (or any) regex engine. Thankfully our matching needs aren't that complicated to begin with :D